test/method.buildCall.js

```diff
@@ -1117,6 +1117,67 @@ describe('lib/web3/method', function () {
                 gasPrice: '23435234234'
             })
         });
+
+        it('should fallback to polling if provider support `on` but `newBlockHeaders` does not arrive in `blockHeaderTimeout` seconds', function (done) {
```

Comment:
This doesn't test if `blockHeaderTimeout` works as expected if a response for `newBlockHeaders` isn't returned, correct?

Reply:
Based on the sequence of assertions provider will expect `eth_getTransactionReceipt` call from the code, that can only happen when `blockHeaderTimeout` triggers and fallback to polling..